com.unity.visualeffectgraph/Editor/Data/VFXData.cs

```diff
@@ -141,6 +141,16 @@ public virtual bool CanBeCompiled()
             // Empty implementation by default
         }
 
+        public void OnRegisteredContextHasBeenAdded(VFXContext context)
+        {
+            if (!m_Owners.Contains(context))
+                throw new InvalidOperationException(string.Format("Unexpected OnRegisterContextHasBeenAdded from context {0} to {1}", context, this));
+
+            var newGraphParent = context.GetGraph();
+            if (newGraphParent != null && m_Parent != newGraphParent)
+                m_Parent = newGraphParent;
```

Comment:
What if the data has other contexts that were not (yet) moved?
Probably we should move only if the data has no parent yet?
There still the potential issue of having contexts from different graph linked together. We should enfore it's not possible